src/JwtCredentialsManager.ts

```diff
@@ -1,27 +1,26 @@
-/* @flow */
-import type { CredentialsManager } from './CredentialsManager';
+import { CredentialsManager } from './CredentialsManager';
 
 type Seconds = number;
 
 export type JwtCredentials = {
-  exp: Seconds,
+  exp: Seconds;
 };
 
-export type JwtCredentialsManagerConfig = {|
-  tokenExpirationMarginSeconds: Seconds | null,
-|};
+export type JwtCredentialsManagerConfig = {
+  tokenExpirationMarginSeconds: Seconds | null;
+};
 
 const SECONDS_TO_MS = 1000;
 
-export default class JwtCredentialsManager<TCredentials: JwtCredentials>
+export default class JwtCredentialsManager<TCredentials extends JwtCredentials>
   implements CredentialsManager<TCredentials> {
   config: JwtCredentialsManagerConfig;
 
-  token: ?string;
+  token: string | null | undefined;
 
-  credentials: ?TCredentials;
+  credentials: TCredentials | null | undefined;
 
-  renewHandle: ?TimeoutID;
+  renewHandle: NodeJS.Timeout | null | undefined;
```

Comment:
Is this the right type? Wasn't sure?

Reply:
it's probably fine. if for some reason we ran in browser, then we'd get a different type, but whatever

Reply:
i hate this type def idiosyncrasy, these are still just numbers AFAIK, but node types have a special type for it

Reply:
it's actually a `Timeout` object in node instead of a number. it's got methods on it and all https://nodejs.org/api/timers.html#timers_class_timeout

Reply:
oo so weird